osu.Server.Spectator/Hubs/Multiplayer/Matchmaking/MatchmakingMatchController.cs

```diff
@@ -459,17 +461,58 @@ private bool anyUsersReady()
             return room.Users.Any(u => u.State == MultiplayerUserState.Ready);
         }
 
-        private bool isMatchComplete()
+        /// <summary>
+        /// Whether the match can be transitioned into a <see cref="MatchmakingStage.Ended">concluded</see> state,
+        /// provided that it is no longer necessary for any users to keep playing.
+        /// </summary>
+        private bool canConcludeMatch()
         {
             return
+                // The match has run through to its natural conclusion.
+                !hasGameplayRoundsRemaining()
                 // No more players left in the room
-                room.Users.Count == 0
-                // Only a single player, that is no longer in gameplay
+                || room.Users.Count == 0
+                // Only a single player remains, that is no longer in gameplay
                 || (anyPlayerQuit && room.Users.Count == 1 && state.Stage != MatchmakingStage.Gameplay)
-                // The match has run through to its natural conclusion.
-                || (state.CurrentRound == totalRounds && state.Stage > MatchmakingStage.Gameplay)
                 // In head-to-head mode, one player has a score that is unattainable by the other.
-                || (state.Users.Count == 2 && state.Users.Any(u => u.Points >= headToHeadMaxPoints));
+                || (isHeadToHead() && state.Users.Any(u => u.Points >= bestOfWinPoints()));
+        }
+
+        /// <summary>
+        /// Whether there are still gameplay rounds to be played.
+        /// If <c>true</c>, users leaving the match will receive an abandon penalty.
+        /// </summary>
+        private bool hasGameplayRoundsRemaining()
+        {
+            return state.CurrentRound < totalRounds() || state.Stage <= MatchmakingStage.Gameplay;
+        }
+
+        /// <summary>
+        /// Whether this room is in head-to-head mode and should use best-of scoring.
+        /// </summary>
+        private bool isHeadToHead()
+        {
+            return state.Users.Count == 2;
```

Comment:
I think there needs to be a `AppSettings.MatchmakingHeadToHeadIsBestOf` usage here still (or somewhere). Or that setting should just be removed?